extensions/scarb-cairo-execute/src/main.rs

```diff
@@ -0,0 +1,243 @@
+use scarb_ui::{OutputFormat, Ui};
+use std::env;
+use std::path::{Path, PathBuf};
+use std::process::ExitCode;
+
+/// Compiles a Cairo project and runs a function marked `#[executable]`.
+/// Exits with 1 if the compilation or run fails, otherwise 0.
+#[derive(Parser, Debug)]
+#[clap(version, verbatim_doc_comment)]
+struct Args {
+    /// Name of the package.
+    #[command(flatten)]
+    packages_filter: PackagesFilter,
+
+    /// The Cairo project path to compile and run.
+    path: PathBuf,
+
+    /// Whether to only build and into the given path.
+    #[clap(long)]
+    build_only: Option<PathBuf>,
+
+    /// Whether to only run a prebuilt executable.
+    #[arg(long, default_value_t = true, conflicts_with = "build_only")]
+    prebuilt: bool,
+
+    #[clap(flatten)]
+    build: BuildArgs,
+
+    #[clap(flatten)]
+    run: RunArgs,
+}
+
+#[derive(Parser, Debug)]
+struct BuildArgs {
+    /// Whether path is a single file.
+    #[arg(long, conflicts_with = "prebuilt")]
+    single_file: bool,
+    /// Allows the compilation to succeed with warnings.
+    #[arg(long, conflicts_with = "prebuilt")]
+    allow_warnings: bool,
+    /// Path to the executable function.
+    #[arg(long, conflicts_with = "prebuilt")]
+    executable: Option<String>,
+}
+
+#[derive(Parser, Debug)]
+struct RunArgs {
+    /// Serialized arguments to the executable function.
+    #[arg(long, value_delimiter = ',', conflicts_with = "build_only")]
+    args: Vec<BigInt>,
+    /// Whether to print the outputs.
+    #[arg(long, default_value_t = false, conflicts_with = "build_only")]
+    print_outputs: bool,
+    /// If set, the program will be run in proof mode.
+    #[clap(long, default_value_t = false, conflicts_with = "build_only", requires_all=["trace_file", "memory_file", "air_public_input", "air_private_input"])]
+    proof_mode: bool,
+    /// If set, the program will be run in secure mode.
+    #[clap(long, conflicts_with = "build_only")]
+    secure_run: Option<bool>,
+    /// The resulting cairo PIE file.
+    #[clap(long, conflicts_with_all = ["proof_mode", "build_only"])]
+    cairo_pie_output: Option<PathBuf>,
+    /// Can we allow for missing builtins.
+    #[clap(long)]
+    allow_missing_builtins: Option<bool>,
+    #[clap(flatten)]
+    proof: ProofModeArgs,
+}
+
+#[derive(Parser, Debug)]
+struct ProofModeArgs {
+    /// The resulting trace file.
+    #[clap(long, conflicts_with = "build_only", requires = "proof_mode")]
+    trace_file: Option<PathBuf>,
+    /// The resulting memory file.
+    #[clap(long, conflicts_with = "build_only", requires = "proof_mode")]
+    memory_file: Option<PathBuf>,
+    /// The resulting AIR public input file.
+    #[clap(long, conflicts_with = "build_only", requires = "proof_mode")]
+    air_public_input: Option<PathBuf>,
+    /// The resulting AIR private input file.
+    #[clap(long, conflicts_with = "build_only", requires = "proof_mode")]
+    air_private_input: Option<PathBuf>,
+}
+
+fn main() -> ExitCode {
```

Comment:
You can rewrite this like in https://github.com/software-mansion/scarb/blob/9f5bb47f52b8bbccb896b7a75482cd6a9493a567/extensions/scarb-doc/src/main.rs#L94-L104 to handle errors in one place.

Reply:
will do